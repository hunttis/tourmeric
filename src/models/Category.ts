export interface Category {
  abbreviation: string;
  formats?: string;
  image?: string;
  imageSmall?: string | null;
  logo: string;
  name: string;
  type: string;
}

export interface UploadedFile {
  downloadURL: string;
  name: string;
}
