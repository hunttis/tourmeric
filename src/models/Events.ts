export interface TourmericEvent {
  category: string;
  createDate: string;
  date: string;
  endDate?: string;
  format: string | null;
  entryFee: string;
  eventType?: string;
  name: string;
  published?: boolean;
  time: string;
  rulesLevel: string;
  playerSlots: string;
  prizes: string;
  link?: string;
  notes: string;
  rules?: string;
}

export interface TourmericEventEntry {
  key: string;
  value: TourmericEvent;
}
