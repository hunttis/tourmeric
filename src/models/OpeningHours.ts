export interface OpeningHoursException {
  date: string;
  name: string;
  openingHours: string;
  status: string;
  open: boolean;
}
