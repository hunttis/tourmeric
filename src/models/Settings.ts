
export interface Settings {
  activeLocationImage?: string;
  activeLogo: string;
  browserTitle?: string;
  companyinfo?: string;
  dateFormat: string;
  features?: Features;
  introText?: string;
  location?: Location;
  openingHours?: OpeningHours;
  pageSubtitle: string;
  pageTitle: string;
  privacyPolicy?: string;
  showSponsors?: boolean;
  subtitleTextColor: string;
  theme: string;
  titleBarAngle: string;
  titleBarColor: string;
  titleBarColor2: string;
  titleBarPercentage: string;
  titleTextColor: string;
  footer?: {[key: string]: Footer};
  locationImage?: string;
}

interface Features {
  events: Feature;
  highlights: Feature;
  storeinfo: Feature;
}

interface Feature {
  active: boolean;
  introtext: boolean;
}

interface Location {
  address: string;
  directions: string;
  email: string;
  phone: string;
}


export interface OpeningHours {
  additionalinfo: string;
  monday: string;
  tuesday: string;
  wednesday: string;
  thursday: string;
  friday: string;
  saturday: string;
  sunday: string;
}

export interface Footer {
  link: string;
  image: string;
  text: string;
}
