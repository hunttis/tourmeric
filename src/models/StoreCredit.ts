export interface TourmericStoreCreditData {
  date: string;
  creditAddedByName: string;
  creditAddedBy: string;
  note: string;
  value: number;
  category: CreditCategories;
}

export interface StoreCreditCategory { [key: string]: string }

export const CATEGORY_RED: CreditCategories = 'red';
export const CATEGORY_YELLOW: CreditCategories = 'yellow';
export const CATEGORY_GREEN: CreditCategories = 'green';
export const CATEGORY_BLUE: CreditCategories = 'blue';
export const CATEGORY_WHITE: CreditCategories = '';

export type CreditCategories = 'yellow' | 'red' | 'green' | 'blue' | '';
