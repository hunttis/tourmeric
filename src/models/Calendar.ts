import { TourmericEvent } from './Events';

export interface Day {
  day?: string;
  dayOfWeek?: number;
  dayName?: string;
  dayString?: string;
  dayLink?: string;
  empty?: boolean;
  eventsForDay?: { key: string, value: TourmericEvent }[];
  ongoingEventsForDay?: { key: string, value: TourmericEvent }[];
}
