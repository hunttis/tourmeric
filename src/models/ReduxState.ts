import { Settings } from './Settings';
import { Category, UploadedFile } from './Category';
import { OpeningHoursException } from './OpeningHours';
import { TourmericEvent } from './Events';
import { TourmericStoreCreditData, StoreCreditCategory } from './StoreCredit';

export interface ReduxState {
  admin: any;
  firebase: FirebaseObject;
  locale: any;
  router: Router;
  editor: Editor;
}

interface Editor {
  returnLocation: string;
}

export interface FirebaseObject {
  data: FirebaseData;
  ordered: OrderedFirebaseData;
  auth: FirebaseAuth;
  profile: FirebaseProfile;
}

export interface Router {
  location: Location;
  action: string;
}

export interface Location {
  pathname: string;
  search: string;
  hash: string;
  key: string;
}

interface OrderedFirebaseData {
  events: [{ key: string, value: TourmericEvent }];
  eventsongoing: [{ key: string, value: TourmericEvent }];
  news: [{ key: string, value: SingleNewsItem }];
  users: [{ key: string, value: User }];
}

interface FirebaseData {
  settings: Settings;
  highlights: { [key: string]: HighLight };
  events: { [key: string]: TourmericEvent };
  eventsongoing: { [key: string]: TourmericEvent };
  categories: { [key: string]: Category };
  participations: { [key: string]: Participation };
  uploadedFiles: { [key: string]: UploadedFile };
  uploadedCategoryLogos: { [key: string]: UploadedFile };
  uploadedHighlightBanners: { [key: string]: UploadedFile };
  uploadedNewsImages: { [key: string]: UploadedFile };
  uploadedFooterItems: { [key: string]: UploadedFile };
  uploadedStoreinfoFiles: { [key: string]: UploadedFile };
  openinghoursexceptions: { [key: string]: OpeningHoursException };
  storecreditcategories: { [key: string]: StoreCreditCategory };
  users: { [key: string]: User };
  news: { [key: string]: SingleNewsItem };
  storecredit: { [key: string]: { [key: string]: TourmericStoreCreditData }};
}

export interface SingleNewsItem {
  createDate: string;
  date: string;
  linkName: string;
  link: string;
  image: string;
  active: boolean;
  name: string;
  text: string;
  summary: string;
}


export interface HighLight {
  active: boolean;
  image: string;
  createDate: string;
  date: string;
  name: string;
}

export interface Participation {
  [key: string]: ParticipationData;
}

export interface ParticipationData {
  date: string;
  firstName: string;
  lastName: string;
  userId: string;
  comment: string;
}

export interface FirebaseAuth {
  uid: string;
  displayName: string;
  photoURL: string;
  email: string | null;
  emailVerified: boolean;
  providerData: ProviderData;
  apiKey: string;
  isEmpty: boolean;
  isLoaded: boolean;
}

export interface ProviderData {
  displayName: string;
  email: string;
  photoURL: string;
  providerId: string;
  uid: string;
}

export interface FirebaseProfile {
  email: string;
  acceptedPrivacyPolicy?: string;
  acceptsNewsLetter?: boolean;
  avatarUrl: string;
  betaParticipation?: any;
  dciNumber: string;
  displayName: string;
  favoriteCategories?: string;
  firstName: string;
  lastName: string;
  landingPage?: PageOption;
  otherEmail?: string;
  providerData: ProviderData[];
  role: string;
  useOtherEmail?: boolean;
  isEmpty: boolean;
  isLoaded: boolean;
}

export type PageOption = 'today' | 'events' | 'storeinfo' | 'userinfo' | 'companyinfo' | 'admintools' | 'admintoolsevents' | 'adminsitesettings' | '';

export interface User {
  avatarUrl?: string;
  displayName?: string;
  providerData?: ProviderData;
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  dciNumber?: string;
  role?: string;
  active: boolean;
}
