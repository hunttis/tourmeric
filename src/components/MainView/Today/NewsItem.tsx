import React, { Component, Fragment } from 'react';
import moment from 'moment';
import { Translate } from 'react-localize-redux';
import _ from 'lodash';
import { SingleNewsItem } from '~/models/ReduxState';

interface Props {
  newsItem: { key: string, value: SingleNewsItem };
  dateFormat: string;
}

interface State {
  activeNewsItem: boolean;
}

export default class NewsItem extends Component<Props, State> {

  state = { activeNewsItem: false }

  setActiveNewsItem(status: boolean) {
    this.setState({ activeNewsItem: status });
  }

  render() {
    const { newsItem, dateFormat } = this.props;
    const newsData = newsItem.value;
    const newsDate = moment(newsData.date, 'YYYY-MM-DD').format(dateFormat);
    const currentActiveNewsItem = this.state.activeNewsItem;
    const formattedContent = newsData.text ? newsData.text.split('\n') : [];
    const longNewsItem = formattedContent.length > 3;
    const footerNeeded = newsData.link || longNewsItem;

    return (
      <Fragment key={newsItem.key}>
        <div className="card">
          <div className="card-header">
            <div className="card-header-title">
              {newsData.name}
            </div>
            <div className="card-header-title">
              {newsDate}
            </div>
          </div>

          <div className="card-content">
            <div className="content">
              {(!longNewsItem || (longNewsItem && currentActiveNewsItem)) && formattedContent.map((paragraph, index) => <p key={`${newsItem.key}-${index}`}>{paragraph}</p>)}

              {(longNewsItem && !currentActiveNewsItem) && formattedContent.map((paragraph, index) => {
                if (index < 2 && !_.isEmpty(paragraph)) {
                  return <p key={`${newsItem.key}-${index}`}>{paragraph}&nbsp;</p>;
                }
                return '';
              })}
              {(longNewsItem && !currentActiveNewsItem) && <p>...<Translate id="continues" /></p>}
            </div>
          </div>

          {footerNeeded &&
            <div className="card-footer">
              {(longNewsItem && !currentActiveNewsItem) &&
                <a className="card-footer-item" onClick={() => this.setActiveNewsItem(true)}><i className="fas fa-caret-square-down" />&nbsp;&nbsp;<Translate id="showfullnewsitem" /></a>
              }
              {currentActiveNewsItem &&
                <a className="card-footer-item" onClick={() => this.setActiveNewsItem(false)}><i className="fas fa-caret-square-up" />&nbsp;&nbsp;<Translate id="showless" /></a>
              }
              {(!newsData.summary && newsData.link) &&
                <a className="card-footer-item" target="_blank" rel="noopener noreferrer" href={newsData.link}><i className="fas fa-external-link-alt" />&nbsp;&nbsp;{newsData.linkName ? newsData.linkName : newsData.link}</a>
              }
            </div>
          }

        </div>
        <div>&nbsp;</div>
      </Fragment>
    );
  }
}
