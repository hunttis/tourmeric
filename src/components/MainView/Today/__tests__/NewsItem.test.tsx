/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import rrf from 'react-redux-firebase';
import NewsItem from '../NewsItem';
import { SingleNewsItem } from '../../../../models/ReduxState';

jest.mock('react-redux-firebase');

describe('News tests', () => {
  beforeEach(() => {
    rrf.isLoaded = jest.fn(() => true);
  });

  it('should render a newsitem', () => {

    rrf.isLoaded = jest.fn(() => false);

    const singleNewsItem: {key: string, value: SingleNewsItem} = {
      key: 'first',
      value: {
        createDate: '1-1-1990',
        date: '31-12-1990',
        linkName: 'Link',
        link: 'http://tosomesite.com',
        image: 'http://someimage/pic.png',
        active: true,
        name: 'First News Item',
        text: 'First\nNews\nItem\nText',
        summary: 'First Short summary',
      },
    };

    const result = shallow(<NewsItem
      newsItem={singleNewsItem}
      dateFormat="DD-MM-YYYY"
    />);

    expect(result).toMatchSnapshot();

    result.setState({ activeNewsItem: true });

    expect(result).toMatchSnapshot();
  });

});
