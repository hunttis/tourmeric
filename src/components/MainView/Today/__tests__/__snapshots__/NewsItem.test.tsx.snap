// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News tests should render a newsitem 1`] = `
<Fragment
  key="first"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header-title"
      >
        First News Item
      </div>
      <div
        className="card-header-title"
      >
        19-12-2031
      </div>
    </div>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        <p
          key="first-0"
        >
          First
           
        </p>
        <p
          key="first-1"
        >
          News
           
        </p>
        <p>
          ...
          <Translate
            id="continues"
          />
        </p>
      </div>
    </div>
    <div
      className="card-footer"
    >
      <a
        className="card-footer-item"
        onClick={[Function]}
      >
        <i
          className="fas fa-caret-square-down"
        />
          
        <Translate
          id="showfullnewsitem"
        />
      </a>
    </div>
  </div>
  <div>
     
  </div>
</Fragment>
`;

exports[`News tests should render a newsitem 2`] = `
<Fragment
  key="first"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <div
        className="card-header-title"
      >
        First News Item
      </div>
      <div
        className="card-header-title"
      >
        19-12-2031
      </div>
    </div>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        <p
          key="first-0"
        >
          First
        </p>
        <p
          key="first-1"
        >
          News
        </p>
        <p
          key="first-2"
        >
          Item
        </p>
        <p
          key="first-3"
        >
          Text
        </p>
      </div>
    </div>
    <div
      className="card-footer"
    >
      <a
        className="card-footer-item"
        onClick={[Function]}
      >
        <i
          className="fas fa-caret-square-up"
        />
          
        <Translate
          id="showless"
        />
      </a>
    </div>
  </div>
  <div>
     
  </div>
</Fragment>
`;
